airbyte-integrations/connectors/source-jira/source_jira/client.py

```diff
@@ -92,3 +90,170 @@ def health_check(self) -> Tuple[bool, str]:
             )
 
         return alive, error_msg
+
+    def _get_issue_keys(self):
```

Comment:
We'll probably want to move away from this method to support very large scales, but this is not necessary right now. Currently this is caching all ticket IDs in memory for re-use (which is efficient from a query standpoint but not from a space standpoint) but for a large organization (say 50+ engineering teams, each with a JIRA project with 1000+ issues) this might start to get large. I think this could carry us a long way though. 

On a separate note, I think this is exposing a potentially big problem with the `BaseClient` abstraction which is: it assumes each stream will be synced in isolation. This is not true for child streams like `issue_comments` or `issue_properties`. The better way to sync those would be to have something like: 

```
if is_selected(parent_stream) or is_selected(parent_stream.child_streams):
  for record in parent_stream:
   if is_selected(parent_stream): 
      yield record

   for child in parent_stream.selected_child_streams: 
       yield from list(child)
```

This way we iterate over the parent stream exactly once and in a streaming manner so we don't have to keep in memory anything that is not strictly necessary.

Reply:
It is good solution.